### PR TITLE
test: add domain_test

### PR DESCRIPTION
**Pull request by Mo-Fatah**
#### What type of PR is this?
test
#### What this PR does / why we need it:
add test cases for `internal/armada/metrics/domain.go` for improving test coverage
#### Special notes for your reviewer:
The test cases are straightforward as the logic of `domain.go` itself, if you think of some corner cases to test I would love to add them.

### CODE REVIEW

File: internal/armada/metrics/domain_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	armadaresource "github.com/armadaproject/armada/internal/common/resource"
+)
+
+func TestRecordJobRuntime_ShouldReportRuntimeDurations(t *testing.T) {
+	jobMetricsRecorder := NewJobMetricsRecorder()
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", time.Second)
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", 2*time.Second)
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", 3*time.Second)
+
+	jobMetricsRecorder.RecordJobRuntime("gpu", "", 2*time.Second)
```
Comment (kannon92) [inline internal/armada/metrics/domain_test.go:18]: I don't understand the labels "cpu" with duration.  It seems weird to me..

Comment (Mo-Fatah) [reply internal/armada/metrics/domain_test.go:18]: yes they can serve as arbitrary pool tag names, but I think when I wrote them I misunderstood the context here. 

> I wonder if this test should also verify with having a couple different priority classes for the same pool?

I believe it will create just another `QueueMetricsRecorder`, for example if we recorded `pool1` with priority classes `1` and `2`, we will have 2 `QueueMetricsRecorder` with different `Metrics`
https://github.com/armadaproject/armada/blob/371fd974cb02ae70fb5c4a54cdedfe8ed487b780/internal/armada/metrics/domain.go#L55-L67

you think we should verify that both metrics are calculated separately ?

File: internal/armada/metrics/domain_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		case "disk":
+			assert.Equal(t, float64(8), queueMetric.Durations.GetMin())
+			assert.Equal(t, float64(10), queueMetric.Durations.GetMax())
+			assert.Equal(t, float64(18), queueMetric.Durations.GetSum())
+			assert.Equal(t, uint64(2), queueMetric.Durations.GetCount())
+		}
+	}
+}
+
+func TestRecordResources_ShouldReportRuntimResources(t *testing.T) {
```
Comment (kannon92) [inline internal/armada/metrics/domain_test.go:52]:
```suggestion
func TestRecordResources_ShouldReportRuntimeResources(t *testing.T) {
```

File: internal/armada/metrics/domain_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	armadaresource "github.com/armadaproject/armada/internal/common/resource"
+)
+
+func TestRecordJobRuntime_ShouldReportRuntimeDurations(t *testing.T) {
+	jobMetricsRecorder := NewJobMetricsRecorder()
+	jobMetricsRecorder.RecordJobRuntime("cpu", "", time.Second)
```
Comment (kannon92) [inline internal/armada/metrics/domain_test.go:14]: Reading this test, I think you are testing three cases in one function. Could you refactor this to be a table driven test?

Comment (Mo-Fatah) [reply internal/armada/metrics/domain_test.go]: can you see now if this what you meant?

**Comment by Mo-Fatah** [comment]
there is a bug actually in the second function, I will fix it now

File: internal/armada/metrics/domain_test.go (hunk 3 above)
Comment (kannon92) [reply internal/armada/metrics/domain_test.go]: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

I think you are just missing a few items but this looks good to me!

Comment (Mo-Fatah) [reply internal/armada/metrics/domain_test.go]: Cool. I tried to make something similar to the article, but I found it a bit tricky in this context.

**Review by richscott — DISMISSED** [review]

**Comment by kannon92** [comment]
@Mo-Fatah looks like something went wrong with this since we last updated.  Can you fix this?  Otherwise it looks good to me!

**Comment by Mo-Fatah** [comment]
> internal/armada/metrics/domain.go

the file I am testing, `internal/armada/metrics/domain.go`, no longer exists. I will see if it is deleted entirely in a previous PR or it was moved elsewhere

**Comment by kannon92** [comment]
> @kannon92 the file I am testing, `internal/armada/metrics/domain.go`, no longer exists in the `metrics` package. I will see if it was deleted entirely in a previous PR or it was moved elsewhere

https://github.com/armadaproject/armada/tree/master/internal/common/metrics maybe?

**Comment by Mo-Fatah** [comment]
> https://github.com/armadaproject/armada/tree/master/internal/common/metrics maybe?

yes it is, pushing the changes now

**Review by kannon92 — APPROVED** [review]
